airflow/www/security.py

```diff
@@ -728,3 +735,102 @@ def check_authorization(
                 return False
 
         return True
+
+    # TODO: Whether to create APISecurityManager and move api related code to it?
+    def is_user_logged_in(self):
+        """Raise if user already logged in"""
+        if g.user is not None and g.user.is_authenticated:
+            raise Unauthenticated(detail="Client already authenticated")  # For security
+
+    def login_with_user_pass(self, username, password):
+        """Convenience method for user login through the API"""
+        self.is_user_logged_in()
+        if self.auth_type not in (AUTH_DB, AUTH_LDAP):
+            raise Unauthenticated(detail="Authentication type do not match")
```

Comment:
Not sure this is the right response, should be a `BadRequest` right? The message also probably needs some adjustment imo.

Reply:
returning Unauthenticated for all failed authentication in the auth endpoint